presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -156,8 +166,81 @@ public void testPlanningFailure()
         assertFailedQuery("SELECT * FROM mock.default.tests_table", "Throw from apply projection");
     }
 
+    @Test
+    public void testAbortedWhileWaitingForResources()
+            throws Exception
+    {
+        Session mySession = Session.builder(getSession())
+                .setSystemProperty("required_workers_count", "17")
+                .setSystemProperty("required_workers_max_wait_time", "10ms")
+                .build();
+        assertFailedQuery(mySession, "SELECT * FROM tpch.sf1.nation", "Insufficient active worker nodes. Waited 10.00ms for at least 17 workers, but only 1 workers are active");
+    }
+
+    @Test
+    public void testKilledWhileWaitingForResources()
+            throws Exception
+    {
+        String testQueryMarker = "test_query_id_" + ThreadLocalRandom.current().nextInt(100_000_000);
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        try {
+            Future<Object> testQueryFuture = executorService.submit(
+                    () -> {
+                        Session mySession = Session.builder(getSession())
+                                .setSystemProperty("required_workers_count", "17")
+                                .setSystemProperty("required_workers_max_wait_time", "5m")
+                                .build();
+                        String sql = format("SELECT nationkey as %s  FROM tpch.sf1.nation", testQueryMarker);
+                        assertFailedQuery(mySession, sql, "Query killed. Message: because");
+                        return null;
+                    });
+
+            Optional<String> queryIdValue = findQueryId(testQueryMarker);
+            assertThat(queryIdValue).as("query id").isPresent();
+
+            getQueryRunner().execute(format("CALL system.runtime.kill_query('%s', 'because') %s", queryIdValue.get(), IGNORE_EVENT_MARKER));
+            testQueryFuture.get(5, TimeUnit.SECONDS);
+        }
+        finally {
+            shutdownAndAwaitTermination(executorService, Duration.ZERO);
+        }
+    }
+
+    private Optional<String> findQueryId(String queryPattern)
+            throws InterruptedException
+    {
+        Optional<String> queryIdValue = Optional.empty();
+        while (queryIdValue.isEmpty()) {
+            Thread.sleep(50);
+            queryIdValue = computeActual(
+                    format("SELECT query_id FROM system.runtime.queries WHERE query LIKE '%%%s%%' AND query NOT LIKE '%%system.runtime.queries%%' %s",
```

Comment:
i think abstaining from `format` would yield to more readable code, since `%` interfere